go-selinux/selinux_linux.go

```diff
@@ -445,6 +469,194 @@ func ComputeCreateContext(source string, target string, class string) (string, e
 	return readWriteCon(filepath.Join(getSelinuxMountPoint(), "create"), fmt.Sprintf("%s %s %d", source, target, classidx))
 }
 
+// catsToBitset stores categories in a bitset.
+func catsToBitset(cats string) (*bitset.BitSet, error) {
+	bitset := &bitset.BitSet{}
+
+	catlist := strings.Split(cats, ",")
+	for _, r := range catlist {
+		ranges := strings.SplitN(r, ".", 2)
+		if len(ranges) > 1 {
+			catstart, err := parseLevelItem(ranges[0], category)
+			if err != nil {
+				return nil, err
+			}
+			catend, err := parseLevelItem(ranges[1], category)
+			if err != nil {
+				return nil, err
+			}
+			for i := catstart; i <= catend; i++ {
+				bitset.Set(uint(i))
+			}
+		} else {
+			cat, err := parseLevelItem(ranges[0], category)
+			if err != nil {
+				return nil, err
+			}
+			bitset.Set(uint(cat))
+		}
+	}
+
+	return bitset, nil
+}
+
+// parseLevelItem parses and verifies that a sensitivity or category are valid
+func parseLevelItem(s string, sep levelItem) (uint32, error) {
+	item := []rune(s)
+	if item[0] != rune(sep) {
+		return 0, ErrLevelSyntax
+	}
+	val, err := strconv.ParseUint(string(item)[1:], 10, 64)
+	if err != nil {
+		return 0, errors.Wrap(err, "failed to parse level")
+	}
+
+	return uint32(val), nil
```

Comment:
Can you elaborate on why you parse the number as uint64, then convert it to uint32, then convert it back to uint[64] (when calling `bitset.Set()`)? If these are supposed to be 32-bit numbers, we should use `strconv.ParseUint(..., 32)` for conversion to catch potential overflows.

Reply:
This was just an oversight after looking at the Go documentation for `ParseUint()`. Following your suggestion for a custom Max/Min, this should also get rid of some of the unnecessary conversions.